app/src/main/java/org/simple/clinic/overdue/communication/CommunicationRepository.kt

```diff
@@ -0,0 +1,87 @@
+package org.simple.clinic.overdue.communication
+
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.simple.clinic.patient.SyncStatus
+import org.simple.clinic.patient.canBeOverriddenByServerCopy
+import org.simple.clinic.sync.SynceableRepository
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.None
+import org.threeten.bp.Instant
+import java.util.UUID
+import javax.inject.Inject
+
+class CommunicationRepository @Inject constructor(
+    val dao: Communication.RoomDao,
+    val userSession: UserSession
+) : SynceableRepository<Communication, CommunicationPayload> {
+
+  fun create(appointmentId: UUID, type: Communication.Type, result: Communication.Result): Completable {
+    return userSession.loggedInUser()
+        .take(1)
+        .doOnNext {
```

Comment:
Is the `doOnNext` operator the right one to use here, since it isn't a side-effect? 

I'd rather do something like:

```kotlin
.flatMap { if(it === None) Single.error() else Single.just(it) }
```